ros/src/twist_controller/dbw_node.py

```diff
@@ -92,6 +105,20 @@ def publish(self, throttle, brake, steer):
         bcmd.pedal_cmd = brake
         self.brake_pub.publish(bcmd)
 
+    def dbw_enabled_cb(self, msg):
+        self.dbw_enabled = msg.data
+        #rospy.logwarn("dbw_enabled_cb: %s", self.dbw_enabled)
+
+        if self.dbw_enabled == True:
+            self.controller.reset()
```

Comment:
Resetting the PID controller seems like a good solution.
What interests me is: can we avoid throwing previous information in order
for PID to work better after we move from manual mode back to autonomous? What do you think?

Reply:
I rearranged the code a bit. It seems to work well switching between manual and automatic mode. Please have a play around and see if it is ok on your end,